sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -1169,7 +1164,15 @@ export class MessageSession extends LinkEntity {
         if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
         delivery.modified(params);
       } else if (operation === DispositionType.deadletter) {
-        delivery.reject(options.error || {});
```

Comment:
deadletter bug fix was not done for sessions and we all missed it, doing that in this PR.

And added more tests for sessions too.